src/Hyperion/Analysis.hs

```diff
@@ -24,37 +27,58 @@ import Hyperion.Internal
 import Hyperion.Measurement
 import Hyperion.Report
 
-data Component = BenchC Text | GroupC Text | SeriesC Text
+data Component = BenchC Text | GroupC Text | SeriesC Int
 
 qualName :: [Component] -> Text
 qualName = go ""
   where
     go index [BenchC txt] = txt <> index
     go index (GroupC txt : comps) = txt <> index <> "/" <> go "" comps
-    go index (SeriesC txt : comps) = go (index <> ":" <> txt) comps
+    go index (SeriesC x : comps) = go (index <> ":" <> Text.pack (show x)) comps
+    go _ _ = error "qualName: Impossible"
+
+qualData :: [Component] -> [Int]
+qualData = go []
+  where
+    go index [BenchC _txt] = index
+    go index (GroupC _txt : comps) = index <> go [] comps
+    go index (SeriesC x : comps) = go (index <> [x]) comps
     go _ _ = error "qualName: Impossible"
 
+dataOf :: Fold Benchmark (Text, [Int])
+dataOf = benchmarkFoldOf (qualName &&& qualData)
+
 namesOf :: Fold Benchmark Text
-namesOf = go []
+namesOf = benchmarkFoldOf qualName
+
+benchmarkFoldOf :: ([Component] -> a) -> Fold Benchmark a
```

Comment:
`Of` already means fold. You don't need to call it `benchmarkFoldOf`. It's a convention documented in `lens`. That said this function looks pretty ad hoc. Best to just inline it at the 2 call sites.